server/src/state.ts

```diff
@@ -21,16 +21,19 @@ import { Diagnostics } from './diagnostics';
 import { LanguageServerState } from './types';
 
 /* browser specific setup code */
-const messageReader = new BrowserMessageReader(self);
-const messageWriter = new BrowserMessageWriter(self);
+let messageReader, messageWriter;
```

Comment:
Why do we need this?

Reply:
We need this to run test cases. 
The self object is a global variable that refers to the worker itself. When running in a Node.js environment npm test, self is not defined, which leads to the error.